packages/auto-approve/test/check-pr.test.ts

```diff
@@ -276,5 +277,223 @@ describe('check pr against config', async () => {
       scopes.done();
       assert.ok(prMatchesConfig);
     });
+
+    it('should correctly identify and pass a PR that requires additional validation with a dependency process', async () => {
+      const validPR = yaml.load(
+        fs.readFileSync(
+          resolve(fixturesPath, 'config', 'valid-schemas', 'valid-schema5.yml'),
+          'utf8'
+        )
+      ) as {rules: checkPR.ValidPr[]};
+
+      const pr = require(resolve(
+        fixturesPath,
+        'events',
+        'pull_request_opened_special_validation_dependency_update'
+      ));
+
+      const scopes = listChangedFilesPR(200, [
+        {
+          filename: 'package.json',
+          sha: '1234',
+          patch:
+            '@@ -1,7 +1,7 @@\n' +
```

Comment:
non-blocking suggestions:
How about to use file based fixtures for these diffs?

Reply:
I think the patches are small enough to be in-line. I also feel it makes debugging a bit simpler. Perhaps if the patches get larger.

Reply:
ok